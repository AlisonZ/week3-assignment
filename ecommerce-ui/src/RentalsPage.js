import React, {Component} from 'react';
import rentals from './_data/airbnbs.json';
import ShoppingCart from './ShoppingCart';
import RentalsList from './RentalsList';
import AddRentalForm from './AddRentalForm';

class RentalsPage extends Component {
 
    constructor() {
        super()
        this.state = {
            selectedRentals: [],
            rentals: rentals,
            showForm: false,
        }
    }

    addToCart = (idx) => {
        return ()=> {
            const selectedRental = this.state.rentals[idx];
            const isASelectedRental = this.state.selectedRentals.includes(selectedRental)
            if( !isASelectedRental) {
                this.setState(prevState => {
                    return {
                        selectedRentals: [...prevState.selectedRentals, selectedRental],
                    };
                });
            }
        }
    }

    removeFromCart = (idx) => {
        const selectedRentalsCopy = [...this.state.selectedRentals];
        selectedRentalsCopy.splice(idx, 1);

        this.setState({
            selectedRentals: selectedRentalsCopy,
        });
    }

    updateRentalState(newRental) {
         this.setState(prevState => {
            return {
                rentals: [...prevState.rentals, newRental],
            };
        });
    }

    addNewRental = (rentalInfo) => {
        const newRental = {
            "title": rentalInfo.title,
            "houseType": rentalInfo.houseType,
            "image": rentalInfo.image,
            "location": {
                "city": rentalInfo.city,
                "country": rentalInfo.country,
            },
            "payment": {
                "cost": parseInt(rentalInfo.addedCost),
                "description": ""
            },
            "host": {
                "name": "Jamie",
                "isSuperhost": true
            },
            "rating": {
                "stars": 4.5,
                "reviews": 381
            }
        };

        this.updateRentalState(newRental);
    }

    toggleShowForm = () => {
        this.setState({showForm: !this.state.showForm});
    }

    render(){        
        return (
            <div>
                <div className="nav-bar-container">
                    <div className="nav-bar-components">
                        <ShoppingCart 
                            selectedRentals={this.state.selectedRentals} 
                            removeFromCart={this.removeFromCart}
                        />

                        <span 
                            onClick={this.toggleShowForm} 
                            className="dark-grey-primary-title add-property"
                        >
                            Add Property
                        </span>
                    </div>
                </div>
                {
                    this.state.showForm ? 
                    <AddRentalForm 
                        onAddNewRental={this.addNewRental}
                        onSubmitToggle={this.toggleShowForm}
                    /> : 
                    null
                }
              
                <RentalsList 
                    rentals={this.state.rentals} 
                    onAddToCart={this.addToCart} 
                    key=""
                />
            </div>
        );
    }
}

export default RentalsPage;
