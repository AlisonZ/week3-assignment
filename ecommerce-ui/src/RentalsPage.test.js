import React from 'react';
import {shallow} from 'enzyme';
import RentalsPage from './RentalsList';
import {MOCK_RENTAL_1, MOCK_RENTAL_2} from './TestMocks';

it('shallow renders Rentals Page', () => {
    const wrapper = shallow(
        <RentalsPage 
        rentals={[MOCK_RENTAL_1, MOCK_RENTAL_2]}
        onAddToCart={()=> {}}
        />);

    expect(wrapper).toMatchSnapshot();
});