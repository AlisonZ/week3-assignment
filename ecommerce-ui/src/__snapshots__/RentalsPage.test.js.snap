// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`shallow renders Rentals Page 1`] = `
<div>
  <h1
    className="top-places-title"
  >
    Top Places to Stay
  </h1>
  <ul
    className="rentals-list unstyled-list"
  >
    <li
      className="rental-component"
      key="0"
    >
      <RentalComponent
        cost={105}
        houseType="Apartment"
        image="https://images.unsplash.com/photo-1490806230066-f7e6f7bf5052?ixlib=rb-0.3.5&ixid=eyJhcHBfaWQiOjEyMDd9&s=2ca889535ea01f912f94ac4ddf0034e0&auto=format&fit=crop&w=500&q=80"
        isSuperhost={true}
        reviews={133}
        stars={3}
        title="Centrally-located Manhattan studio"
      />
      <button
        className="add-to-cart-btn"
      >
        Add to Cart
      </button>
    </li>
    <li
      className="rental-component"
      key="1"
    >
      <RentalComponent
        cost={130}
        houseType="Apartment"
        image="https://images.unsplash.com/photo-1519424187720-db6d0fc5a5d2?ixlib=rb-0.3.5&ixid=eyJhcHBfaWQiOjEyMDd9&s=19f783ae564d7c6da4777cde1a84c1c9&auto=format&fit=crop&w=500&q=80"
        isSuperhost={true}
        reviews={381}
        stars={4.5}
        title="Denver apartment outside Union Station"
      />
      <button
        className="add-to-cart-btn"
      >
        Add to Cart
      </button>
    </li>
  </ul>
</div>
`;
